tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Kinesis/AwsKinesisCommon.cs

```diff
@@ -38,6 +38,17 @@ internal static class AwsKinesisCommon
             return arnComponents[1];
         }
 
+        public static string? GetStreamName(IAmazonKinesisRequestWithStreamNameAndStreamArn request)
+        {
+            string? streamName = StreamNameFromARN(request.StreamARN);
+            if (StringUtil.IsNullOrEmpty(streamName))
+            {
+                streamName = request.StreamName;
+            }
+
+            return streamName;
```

Comment:
We could simplify this:
```suggestion
            string? streamName = StreamNameFromARN(request.StreamARN);
            return StringUtil.IsNullOrEmpty(streamName)? request.StreamName : streamName ;
```

At least I find it a bit less confusing this way, personally